src/main/java/com/onarandombox/MultiverseCore/commands/DumpsCommand.java

```diff
@@ -0,0 +1,320 @@
+                                types.add(type.name().toLowerCase());
+                            }
+                            return types;
+                        })
+                        .build())
+                .add(CommandValueFlag.builder("--upload", Services.class)
+                        .addAlias("-u")
+                        .context((value) -> {
+                            try {
+                                return Services.valueOf(value.toUpperCase());
+                            } catch (IllegalArgumentException e) {
+                                throw new InvalidCommandArgument("Invalid service " + value + " in --upload");
+                            }
+                        })
+                        .completion(() -> {
+                            List<String> types = new ArrayList<>();
+                            for (Services type : Services.values()) {
+                                types.add(type.name().toLowerCase());
+                            }
+                            return types;
+                        })
+                        .build())
+                .add(CommandFlag.builder("--paranoid")// Does not upload logs or plugin list (except if --logs mclogs is there)
+                        .addAlias("-p")
+                        .build())
+                .build());
+    }
+
+    private enum Services {
+        PASTEGG,
+        PASTESDEV
+    }
+
+    private enum LogsType {
+        APPEND,
+        MCLOGS
+    }
+
+    @Subcommand("dumps")
+    @CommandPermission("multiverse.core.dumps")
+    @CommandCompletion("@flags:groupName=mvdumps")
+    @Syntax("[--logs <mclogs|append>] [--upload <pastesdev|pastegg>] [--paranoid]")
+    @Description("{@@mv-core.dumps.description}")
+    public void onDumpsCommand(CommandIssuer issuer,
+
+                               @Optional
+                               @Syntax("[--logs <mclogs|append>] [--upload <pastesdev|pastegg>] [--paranoid]")
+                               String[] flags
+    ) {
+        ParsedCommandFlags parsedFlags = parseFlags(flags);
+
+        MVVersionEvent versionEvent = new MVVersionEvent();
+
+        this.addDebugInfoToEvent(versionEvent);
+        plugin.getServer().getPluginManager().callEvent(versionEvent);
+
+        // Add plugin list if user isn't paranoid
+        if (!parsedFlags.hasFlag("--paranoid")) {
+            versionEvent.putDetailedVersionInfo("plugins.md", "# Plugins\n\n" + getPluginList());
+        }
+
+
+        // Deal with default case of logs
+        LogsType logsType = parsedFlags.flagValue("--logs", LogsType.class);
+        if (logsType == null) {
+            logsType = LogsType.MCLOGS;
+            Logging.finer("Logs was null so set to mclogs");
+        }
+
+
+
+
+        // Deal with default case of upload
+        Services services = parsedFlags.flagValue("--upload", Services.class);
+        if (services == null) {
```

Comment:
set a defaultValue for command flag, so you dont need null checking

Reply:
ohhh right you can do `parsedFlags.flagValue("--upload", ServiceOption.PASTEGG, ServiceOption.class)`